### PR TITLE
Grab information on examine, no message spam when removing cuffs

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR adds to examine text information if person is grabbing someone
This PR also prevents you from starting removing cuffs/unbuckle multiple times

## Why It's Good For The Game

Seeing if person is strangling guy under them by examining is good as you supposed to see that they are strangling them, you cant do it stealthy
Preventing you from spamming "trying to unbuckle yourself" and same with cuffs and muzzle is good, spamming chat is bad.

## Images of changes

image

## Testing

Compiled, tried some examining

## Changelog
:cl:
tweak: If you grab a person, that would be shown on examine.
fix: People no longer spam chat with unbuckling/removing cuffs multiple times
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/examine.dm [hunk 1]
```diff
@@ -134,6 +136,8 @@
 		if(length(parts) >= 5)
 			accessories = parts[5]
 
+		if(item && istype(item, /obj/item/grab)
```
Comment (DGamerL) [inline code/modules/mob/living/carbon/examine.dm:139]:
```suggestion
		if(item && istype(item, /obj/item/grab))
```

File: code/modules/mob/living/carbon/examine.dm [hunk 2]
```diff
@@ -168,6 +172,18 @@
 		else
 			msg += "<span class='warning'>[p_they(TRUE)] [p_are()] [bicon(legcuffed)] legcuffed!</span>\n"
 
+	for(var/obj/item/grab/grab in grab_items)
+		switch(grab.state)
+			if(GRAB_AGGRESSIVE)
+				msg += "<span class='boldwarning'>[p_they(TRUE)] [p_are()] holding [grab.affecting]'s hands!</span>\n"
+			if(GRAB_NECK)
+				msg += "<span class='boldwarning'>[p_they(TRUE)] [p_are()] holding [grab.affecting] neck!</span>\n"
```
Comment (DGamerL) [inline code/modules/mob/living/carbon/examine.dm:180]:
```suggestion
				msg += "<span class='boldwarning'>[p_they(TRUE)] [p_are()] holding [grab.affecting]'s neck!</span>\n"
```

**Review by DGamerL — COMMENTED** [review]
First look

**Comment by Contrabang** [comment]
I think escaping cuffs/legcuffs/etc examines shouldn't be added, If a person manages to hide their visible message of trying to escape in chaos, they should be rewarded for it. If a prisoner escapes, and all of sec leaves processing to catch them, the person still inside processing should be allowed to capitalize on their captor's neglect

**Comment by matttheficus** [comment]
Objecting on the cuffs/legcuffs/buckled notices - you should be rewarded for choosing the right time to do so.

**Comment by procdrone** [comment]
for gameplay reasons, handcuff interactions shouldn't be displayed

**Comment by HMBGERDO** [comment]
okay, i removed examine information about removing handcuffs and such
grab info still here, no spam "trying to unbuckle" also here

**Comment by DGamerL** [comment]
> okay, i removed examine information about removing handcuffs and such grab info still here, no spam "trying to unbuckle" also here

What do you mean with spam?

**Comment by HMBGERDO** [comment]
> What do you mean with spam?

Previously, if you click resist button multiple times while cuffed/buckled, you will start unbuckling every time you press button with new progressbar and new red message in chat
Now you cant do it

**Comment by DGamerL** [comment]
> > What do you mean with spam?
> 
> Previously, if you click resist button multiple times while cuffed/buckled, you will start unbuckling every time you press button with new progressbar and new red message in chat Now you cant do it

Very nice. How does it handle if you're in the process of uncuffing yourself, but get buckled in a chair again?

**Comment by HMBGERDO** [comment]
> Very nice. How does it handle if you're in the process of uncuffing yourself, but get buckled in a chair again?

if you're asking me, would remove handcuffs progressbar unbuckle you - no, this was not changed, this works like before
if you're asking me, could you start unbuckling while removing cuffs bar is not gone - yes, there is different action defines, if you're trying to unbuckle - removing cuffs action wont stop you. Removing cuffs while unbuckling is possible, removing cuffs while also removing cuffs - impossible

**Comment by DGamerL** [comment]
> > Very nice. How does it handle if you're in the process of uncuffing yourself, but get buckled in a chair again?
> 
> Removing cuffs while unbuckling is possible

I assume you mean that uncuffing still fails if the person is buckled again when the timer runs out?

**Comment by SteelSlayer** [comment]
PR has been open for a month and has 3 objections, no one in favor. Closing.

**Review by S34NW — CHANGES_REQUESTED** [review]
Please update your PR description to properly reflect the changes to the PR where you removed the unbuckling examine text. This has caused great delays in a team vote on this PR.
Also please deconflict it. Once these have been done it is good for merge.

**Review by S34NW — APPROVED** [review]

**Comment by Scribble-Sheep** [comment]
Would the change to prevent spam clicking affect the current behavior with being stunned while trying to resist out of bucklecuffs?

As of right now, if you're stunned in progress, the progress bar remains, but won't do anything when finished, so you have to start a new one. If this prevents you from starting a new one, but doesn't remove the existing "nonfunctional" progress bar, that seems like it could be a slight security buff, as you'll have to wait out the timer of the old one before resisting again.

**Comment by HMBGERDO** [comment]
> Would the change to prevent spam clicking affect the current behavior with being stunned while trying to resist out of bucklecuffs?
> 
> As of right now, if you're stunned in progress, the progress bar remains, but won't do anything when finished, so you have to start a new one. If this prevents you from starting a new one, but doesn't remove the existing "nonfunctional" progress bar, that seems like it could be a slight security buff, as you'll have to wait out the timer of the old one before resisting again.

Wait what
If you're stamcritted while removing cuffs, progress bar should be gone
If you're not stamcritted then bar should stay and have effect when completed

Are you sure what you're saying is right? Gonna test it tomorrow

I know only one case when progressbar ends and does nothing, if you start removing cuffs and then get buckled on the same place, then at the end you won't remove cuffs nor unbuckle

**Comment by Scribble-Sheep** [comment]
> > Would the change to prevent spam clicking affect the current behavior with being stunned while trying to resist out of bucklecuffs?
> > As of right now, if you're stunned in progress, the progress bar remains, but won't do anything when finished, so you have to start a new one. If this prevents you from starting a new one, but doesn't remove the existing "nonfunctional" progress bar, that seems like it could be a slight security buff, as you'll have to wait out the timer of the old one before resisting again.
> 
> Wait what If you're stamcritted while removing cuffs, progress bar should be gone If you're not stamcritted then bar should stay and have effect when completed
> 
> Are you sure what you're saying is right? Gonna test it tomorrow
> 
> I know only one case when progressbar ends and does nothing, if you start removing cuffs and then get buckled on the same place, then at the end you won't remove cuffs nor unbuckle

I mean, I remember getting stunbatonned while bucklecuffed has that as well? Certainly happens to me a lot, but now I'm second guessing myself.
At the very least, it's worth testing as an edge case for something like this.

**Comment by HMBGERDO** [comment]
alright, tests:
1:
standing still, removing cuffs, getting batonned
result: fallen on the floor, progress bar stays, at the end cuffs removed

2:
standing still, removing cuffs, getting batonned twice(stamcrit)
result: when critted, progress bar gone

3:
being buckled, unbuckling, getting single baton
result: progress bar stays, unbuckled

4: 
being buckled, unbuckling, stamcrit
result: progress bar gone

works as intended

File: code/modules/mob/living/carbon/carbon_defines.dm [hunk 3]
```diff
@@ -28,3 +28,5 @@
 	var/nightmare = 0
 
 	blood_volume = BLOOD_VOLUME_NORMAL
+
+	var/list/current_actions = list()
```
Comment (lewcc) [inline code/modules/mob/living/carbon/carbon_defines.dm:32]: I personally feel like a status effect might be a better solution than a new global list for each and every human. That's just my view, though, and I'd like others to take another look at this and see what they think.

**Review by lewcc — COMMENTED** [review]
Just one thing -- might be fine, but I would like someone else to take another look.

File: code/modules/mob/living/carbon/carbon_defines.dm (hunk 3 above)
Comment (HMBGERDO) [reply code/modules/mob/living/carbon/carbon_defines.dm:32]: well, status effects are also stored in a list on /mob/living so i dont get the argument but sure, i will do that with status effects

Comment (HMBGERDO) [reply code/modules/mob/living/carbon/carbon_defines.dm:32]: done

File: code/datums/status_effects/neutral.dm [hunk 4]
```diff
@@ -270,3 +270,19 @@
...
+	id= "remove handcuffs"
+
+/datum/status_effect/action_status_effect/break_handcuffs
+	id= "break handcuffs"
+
+/datum/status_effect/action_status_effect/remove_muzzle
+	id= "remove muzzle"
+
+/datum/status_effect/action_status_effect/unbuckle
+	id= "unbuckle"
```
Comment (hal9000PR) [inline code/datums/status_effects/neutral.dm:288]: instead of creating a bunch of empty status effects, couldnt you register a signal to listen for when someone gets examined when they start removing cuffs, muzzle, etc. and when it hears the signal show a message to the examiner?

Comment (hal9000PR) [inline code/datums/status_effects/neutral.dm:288]:
```suggestion
/datum/status_effect/action_status_effect
	alert_type = null
	tick_interval = -1

/datum/status_effect/action_status_effect/remove_handcuffs
	id = "remove_handcuffs"

/datum/status_effect/action_status_effect/break_handcuffs
	id = "break_handcuffs"

/datum/status_effect/action_status_effect/remove_muzzle
	id = "remove_muzzle"

/datum/status_effect/action_status_effect/unbuckle
	id = "unbuckle"
```
